pkg/kubelet/gpu/nvidia/nvidia.go

```diff
@@ -0,0 +1,132 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nvidia
+
+import (
+	gpuutil "k8s.io/kubernetes/pkg/kubelet/gpu/nvidia/util"
+	gputypes "k8s.io/kubernetes/pkg/kubelet/gpu/types"
+)
+
+const (
+	Vendor string = "NVIDIA"
+	// All NVIDIA GPUs cards should be mounted with nvidiactl and nvidia-uvm
+	NvidiaDeviceCtl string = "/dev/nvidiactl"
+	NvidiaDeviceUVM string = "/dev/nvidia-uvm"
+)
+
+// TODO: If use NVML in the future, the implementation could be more complex,
+// and more powerful!
+
+type NvidiaGPU struct {
+	gpuInfo []gputypes.GPUDevice
+}
+
+func ProbePlugin() (gputypes.GPUPlugin, error) {
+	var nvidiaGPU NvidiaGPU
+
+	err := nvidiaGPU.InitPlugin()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &nvidiaGPU, nil
+}
+
+// Get the vendor information.
+func (nvidiaGPU *NvidiaGPU) Vendor() string {
+	return Vendor
+}
+
+func (nvidiaGPU *NvidiaGPU) InitPlugin() error {
+	allPaths, err := gpuutil.Discovery()
+
+	if err != nil {
+		return err
+	}
+
+	// Initialize the Nvidia device information.
+	for _, path := range allPaths {
+		nvidiaGPU.gpuInfo = append(nvidiaGPU.gpuInfo, gputypes.GPUDevice{Path: path, Status: gputypes.GPUFree, ContainerID: ""})
+	}
+
+	return nil
+}
+
+func (nvidiaGPU *NvidiaGPU) ReleasePlugin() error {
+	nvidiaGPU.gpuInfo = nil
+
+	return nil
+}
+
+func (nvidiaGPU *NvidiaGPU) Capacity() (int, error) {
+	return len(nvidiaGPU.gpuInfo), nil
```

Comment:
Should this and the other methods be protected with a mutex for concurrent accesses?